src/componentWithName.js

```diff
@@ -16,7 +16,7 @@ function hasName(name, prop, propName, componentName) {
   }
 
   const { type } = prop;
-  if (getComponentName(type) !== name) {
+  if (!new RegExp(name).test(getComponentName(type))) {
```

Comment:
`componentWithName('[Bracketed]Foo')` with your PR as-is would suddenly start matching `BFoo`, `rFoo`, `aFoo`, `cFoo`, etc, where previously it only matched the literal `[Bracketed]Foo` - which is a semver-major change.